libs/src/main/python/dlpx/virtualization/libs/libs.py

```diff
@@ -410,3 +415,30 @@ def _log_request(message, log_level):
 
     response = internal_libs.log(log_request)
     _handle_response(response)
+
+
+def retrieve_credentials(credentials_supplier):
```

Comment:
can we add tests for this in test_libs.py? I think you could maybe use the other tests in that file as an example?